packages/playwright/src/index.ts

```diff
@@ -672,7 +672,12 @@ class ArtifactsRecorder {
   }
 
   private _attachScreenshot(screenshotPath: string) {
-    this._testInfo.attachments.push({ name: 'screenshot', path: screenshotPath, contentType: 'image/png' });
+    const step = this._testInfo._addStep({
```

Comment:
As per discussion at the meeting, this code will stay the way it is, no need to change first and return to what it was, just revert.

Reply:
I don't recall discussing this in the meeting. I don't think it can stay exactly the way it is because we'd lose the `attach "screenshot"` step then. That step is crucial because without it, it won't be shown in Trace Viewer (see LINK TO BE ADDED). I've migrated it to use `attach()` in https://github.com/microsoft/playwright/pull/34037/commits/bc66311a05cb0925b8b970b8c5b4b5db56e375fe, which is much smaller of a change and retains the current behaviour.

Reply:
now that i've implemented a workaround, scratch that. reverted!